lib/src/modules/gates.dart

```diff
@@ -513,3 +513,64 @@ class Mux extends Module with InlineSystemVerilog {
     return '$control ? $d1 : $d0';
   }
 }
+
+/// A generic two-input bit index gate [Module].
+///
+/// It always takes two inputs and has one output of width 1.
+class IndexGate extends Module with InlineSystemVerilog {
+  late final String _a, _b, _y;
+
+  /// The primary input to this gate.
+  Logic get a => input(_a);
+
+  /// The bit index for this gate.
+  Logic get b => input(_b);
+
+  /// The output of this gate.
+  Logic get y => output(_y);
```

Comment:
Let's name these inputs and output signals better.  I know there's some poor naming on some other generic gates, but this one has very specific meaning.  Maybe `original`, `index`, and `selection` or something instead of a, b, y?